### PR TITLE
wasm: make dependency clearer

### PR DESCRIPTION
**Pull request by lizan**
Partially addresses #12574, makes split the monolithic wasm_lib into multiple libraries. Clear up dependency metadata due to select.

Risk Level: Low
Testing: CI

Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: source/extensions/common/wasm/wasm_runtime_factory.h [hunk 1]
```diff
@@ -2,13 +2,14 @@
 
 #include <string_view>
 
-#include "extensions/common/wasm/wasm_vm.h"
-
+#include "include/proxy-wasm/wasm_vm.h"
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/wasm_runtime_factory.h:5]: Empty line was unnecessarily removed here.

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -936,17 +885,7 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         strip_prefix = "rules_rust-{version}",
         urls = ["https://github.com/bazelbuild/rules_rust/archive/{version}.tar.gz"],
         use_category = ["dataplane_ext"],
-        extensions = [
-            "envoy.access_loggers.wasm",
-            "envoy.bootstrap.wasm",
-            "envoy.filters.http.wasm",
-            "envoy.filters.network.wasm",
-            "envoy.stat_sinks.wasm",
```
Comment (PiotrSikora) [inline bazel/repository_locations.bzl:944]: `rules_rust` is also used for building tests.

**Review by PiotrSikora — COMMENTED** [review]
Looks great, thanks! LGTM, sans comments.

File: bazel/repository_locations.bzl (hunk 2 above)
Comment (lizan) [reply bazel/repository_locations.bzl:944]: test dependencies are not counted, this used to be `test_only` before we added wasmtime.

Comment (mathetake) [reply bazel/repository_locations.bzl:944]: FYI, https://github.com/envoyproxy/envoy/pull/13932/commits/3f62cc67141c6f3d229bc70ed737073087ca10ac#diff-a2ca502474bb8ec5f092176f7ee6d2f11fc395ebb00e1717d6fbad9b16ee8fd8L905 this commit changes `test_only` to `dataplane_ext`

Comment (mathetake) [reply bazel/repository_locations.bzl:944]: in wasmtime PR

**Review by mathetake — DISMISSED** [review]
Thank you!

**Review by PiotrSikora — DISMISSED** [review]

**Review by htuch — APPROVED** [review]
Much clearer, thanks!

**Pull request merged**